File: structural/bridge/bridge.js
class Color {
  fill() {}
}

class Blue extends Color {
  fill() {
    return "Color is Blue";
  }
}

class Red extends Color {
  fill() {
    return "Color is Red";
  }
}

class Shape {
  constructor(color) {
    this.color = color;
  }

  draw() {}
}

class Square extends Shape {
  constructor(color) {
    super(color);
  }

  draw() {
    return "Square drawn. " + this.color.fill();
  }
}

class Triangle extends Shape {
  constructor(color) {
    super(color);
  }

  draw() {
    return "Triangle drawn. " + this.color.fill();
  }
}

//a square with red color
const square = new Square(new Red());
console.log(square.draw());

//a triangle with blue color
const triangle = new Triangle(new Blue());
console.log(triangle.draw());

File: structural/bridge/bridge.py
class Color:
    def fill(self):
        pass


class Blue(Color):
    def fill(self):
        return "Color is Blue"


class Red(Color):
    def fill(self):
        return "Color is Red"


class Shape:
    def __init__(self, color):
        self.color = color

    def draw(self):
        pass


class Square(Shape):
    def __init__(self, color):
        super().__init__(color)

    def draw(self):
        return "Square drawn. " + self.color.fill()


class Triangle(Shape):
    def __init__(self, color):
        super().__init__(color)

    def draw(self):
        return "Triangle drawn. " + self.color.fill()


# a square with red color
square = Square(Red())
print(square.draw())

# a triangle with blue color
triangle = Triangle(Blue())
print(triangle.draw())
